ShopifySharp/Entities/CarrierShippingRateProvider.cs

```diff
@@ -0,0 +1,37 @@
+using System.Collections.Generic;
+using Newtonsoft.Json;
+
+namespace ShopifySharp
+{
+    public class CarrierShippingRateProvider : ShopifyObject
+    {
+        /// <summary>
+        /// A Carrier Service (also known as a Carrier Calculated Service or Shipping Service) provides real-time shipping rates to Shopify. Some common carrier services include: Canada Post, FedEx, UPS, and USPS. Note that the term "carrier" is often used interchangeably with the terms "shipping company" and "rate provider."
+        /// </summary>
+        [JsonProperty("carrier_service_id")]
+        public long? CarrierServiceId { get; set; }
+
+        /// <summary>
+        /// Rate adjustments - Flat fee
+        /// </summary>
+        [JsonProperty("flat_modifier")]
+        public string FlatModifier { get; set; }
+
+        /// <summary>
+        /// Rate adjustments - percentage
+        /// </summary>
+        [JsonProperty("percent_modifier")]
+        public int PercentModifier { get; set; }
+
+
+        [JsonProperty("service_filter")]
+        public ServiceFilter ServiceFilter { get; set; }
```

Comment:
I mentioned below that I lean toward removing the `ServiceFilter` class since it has no documentation. If we do that I'd change this property and any others that use the ServiceFilter class to an `object` type. If devs are using this prop they'll probably know what they're looking for here.